flink-java/src/main/java/org/apache/flink/api/java/ClosureCleaner.java

```diff
@@ -101,6 +132,29 @@ public static void clean(Object func, boolean checkSerializable) {
 		}
 	}
 
+	public static void clean(Object func, boolean checkSerializable) {
+		clean(func, ExecutionConfig.CLEANLEVEL.RECURSION, checkSerializable);
+	}
+
+	private static boolean isNeedToDig(Field f, Object fo, Class<?> cls) {
```

Comment:
I think this should be called `needsRecursion` or something like it that makes the recursion clearer.